lms/djangoapps/course_home_api/utils.py

```diff
@@ -0,0 +1,20 @@
+""" Utilities for views in the course home api"""
+from rest_framework.exceptions import PermissionDenied
+
+from lms.djangoapps.courseware.courses import get_course_with_access as base_get_course_with_access
+from lms.djangoapps.courseware.exceptions import CourseAccessRedirect
+
+
+def get_course_with_access_avoid_redirect(*args, **kwargs):
```

Comment:
#NIT I don't like how the function is named. I suggest `get_course_or_impermissible_error` or something like that.